src/ipc/ipc-common/IPCProxyBase.h

```diff
@@ -101,6 +100,30 @@ class IPCProxyBase : public AdapterType, protected IPCProxyBaseBase
         QPointer<ProxyType> m_proxy;
         IPCProxyBase &m_owner;
     };
+    /**
+     * Call the given function once, as soon as the proxy reaches the "ready" state
+     * If the proxy is already in "ready" state, the function is called immediately, synchronously
+     */
+
+    template<typename Function>
+    void callOnceReady(const QObject* context, Function f)
+    {
+        if (ready()) {
+            f();
+        }
+        else {
+            auto connection = std::make_unique<QMetaObject::Connection>();
+            *connection = QObject::connect(this, &InterfaceBase::componentCompleted,
+                                           context, [this, f{std::move(f)}, connection{std::move(connection)} ]()
+                {
+                    if (this->ready()) {
+                        f();
+                        QObject::disconnect(*connection);
+                    }
+                }
+            );
+        }
+    }
```

Comment:
The whole idea was to NOT add it this way but to create an external observer. This causes this class to grow.

Reply:
Such a helper is useful for any class which extends InterfaceBase, not only for IPCProxies.

Reply:
> Such a helper is useful for any class which extends InterfaceBase, not only for IPCProxies.

exactly. That is why it should be added as external observer class, not inside just one class.